src/main/java/com/github/manolo8/darkbot/core/objects/facades/SlotBarsProxy.java

```diff
@@ -54,40 +58,56 @@ public boolean isProActionBarVisible() {
-        return ((Item) item).hasShortcut();
+    public Optional<eu.darkbot.api.items.Item> getItem(@NotNull SelectableItem selectableItem) {
+        return (Optional<eu.darkbot.api.items.Item>) findItem(selectableItem);
     }
 
     @Override
-    public boolean selectItem(@NotNull eu.darkbot.api.objects.Item item) {
-        SlotBarsProxy.Type slotBarType = ((Item) item).getSlotBarType();
-        int slotNumber = ((Item) item).getFirstSlotNumber();
-
-        if (slotBarType == null || slotNumber == -1 ||
-                (slotBarType == SlotBarsProxy.Type.PRO_ACTION_BAR &&
-                        !isProActionBarVisible() &&
-                        !settings.pressKeybind(SettingsProxy.KeyBind.TOGGLE_PRO_ACTION)))
-            return false; //return false if slot type is pro action bar & isn't visible & keybind wasn't toggled.
-
-        return settings.pressKeybind(SettingsProxy.KeyBind.of(slotBarType, slotNumber));
+    public Collection<? extends eu.darkbot.api.items.Item> getItems(@NotNull ItemCategory itemCategory) {
+        return categoryBar.get(itemCategory).items;
     }
 
     @Override
-    public Collection<? extends eu.darkbot.api.objects.Item> getItems() {
-        return categoryBar.categories.stream()
-                .flatMap(category -> category.items.stream())
-                .collect(Collectors.toList());
+    public Optional<eu.darkbot.api.items.Item> getAvailable(@NotNull SelectableItem selectableItem) {
+        return getItem(selectableItem)
+                .filter(eu.darkbot.api.items.Item::isAvailable)
+                .filter(i -> ((Item) i).hasShortcut());
     }
 
     @Override
-    public boolean hasCategory(@NotNull Category category) {
-        return categoryBar.hasCategory(category);
+    public @NotNull ItemUseResult useItem(@NotNull SelectableItem selectableItem, ItemUseFlag... itemFlags) {
+        Item item = (Item) getAvailable(selectableItem).orElse(null);
+        if (item == null) return ItemUseResult.NOT_AVAILABLE;
+
+        for (ItemUseFlag useCase : itemFlags)
+            if (!useCase.test(item))
+                return useCase.getFailResult();
+
+        SlotBarsProxy.Type slotBarType = item.getSlotBarType();
+        int slotNumber = item.getFirstSlotNumber();
+        if (slotNumber < 1 || slotNumber > 10) return ItemUseResult.FAILED;
+
+        boolean toggleProAction = (slotBarType == SlotBarsProxy.Type.PRO_ACTION_BAR) != isProActionBarVisible();
+
+        return (!toggleProAction || settings.pressKeybind(SettingsProxy.KeyBind.TOGGLE_PRO_ACTION))
+                && settings.pressKeybind(SettingsProxy.KeyBind.of(slotBarType, slotNumber))
+                ? ItemUseResult.SUCCESS : ItemUseResult.FAILED;
+
     }
 
-    @Override
-    public Collection<? extends eu.darkbot.api.objects.Item> getItems(@NotNull Category category) {
-        return categoryBar.get(category).items;
+    private Optional<? extends eu.darkbot.api.items.Item> findItem(SelectableItem item) {
```

Comment:
i'm pretty sure `get` is more common in the java world, and should probably be prefered over `find` since in this case it fits just as well